home/app_manager.py

```diff
@@ -90,6 +92,24 @@ class AppManagerWidget(ipw.VBox):
         </div>"""
     )
 
+    CORE_DEPENDENCIES_COMPATIBILTIY_WARNING = Template(
+        """<div class="alert alert-danger">
+    The version to be install of this app is not compatible with this AiiDAlab environment.
```

Comment:
Maybe like this is a bit better.
```suggestion
    The to-be-installed version of this app is not compatible with this AiiDAlab environment.
```

Reply:
I add a commit https://github.com/aiidalab/aiidalab-home/pull/130/commits/7a87dca3654c236dd926a2171caa561ac49db8b6 to address this comment. But I find it can be simplified by what I mentioned having a more versatible `Dependency` dataclass, then I do not need to do the core package compatible check in the widget here but simply check with the property of the Dependency object.